FullClusterRestartIT.testSnapshotRestore fails on 6.x

```
FAILURE 13.1s | FullClusterRestartIT.testSnapshotRestore <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: template doesn't match:
   >                  test_template: 
   >                          aliases: 
   >                             alias1: same [empty map]
   >                             alias2: 
   >                               filter: 
   >                                   term: 
   >                                  version: same [6.0.0]
   >                         mappings: 
   >                                doc: 
   >                              _source: 
   >                                enabled: same [true]
   >                            order: same [0]
   >                         settings: 
   >                              index: 
   >                     number_of_shards: same [1]
   >                         template: expected [evil_*] but not found
   >                   index_patterns: unexpected but found [[evil_*]]
   > 	at __randomizedtesting.SeedInfo.seed([B9588B59D97307DD:863D61A013CE00A2]:0)
   > 	at org.elasticsearch.upgrades.FullClusterRestartIT.checkSnapshot(FullClusterRestartIT.java:876)
   > 	at org.elasticsearch.upgrades.FullClusterRestartIT.testSnapshotRestore(FullClusterRestartIT.java:760)
   > 	at java.lang.Thread.run(Thread.java:745)
```

Reproduces with:

```
gradle :qa:full-cluster-restart:v6.0.0-SNAPSHOT#oldClusterTestRunner -Dtests.seed=B9588B59D97307DD -Dtests.class=org.elasticsearch.upgrades.FullClusterRestartIT -Dtests.method="testSnapshotRestore" -Dtests.security.manager=true -Dtests.locale=ar-QA -Dtests.timezone=America/Anguilla
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.x+intake/571/consoleFull

--- a/qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java
+++ b/qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java
@@ -857,7 +857,7 @@ public class FullClusterRestartIT extends ESRestTestCase {
         // Check that the template was restored successfully
         map = toMap(client().performRequest("GET", "/_template/test_template"));
         expected = new HashMap<>();
-        if (runningAgainstOldCluster) {
+        if (runningAgainstOldCluster && oldClusterVersion.before(Version.V_6_0_0_beta1)) {
             expected.put("template", "evil_*");
         } else {
             expected.put("index_patterns", singletonList("evil_*"));

